src/player_avatar.c

```diff
@@ -0,0 +1,731 @@
+    PlayerAvatar *avatar = Heap_Alloc(HEAP_ID_FIELD2, sizeof(PlayerAvatar));
+    GF_ASSERT(avatar != NULL);
+    memset(avatar, 0, sizeof(PlayerAvatar));
+    return avatar;
+}
+
+static void PlayerAvatar_Setup(PlayerAvatar *avatar, s32 state, u32 gender, PlayerSaveData *playerSaveData) {
+    PlayerAvatar_SetPlayerSaveData(avatar, playerSaveData);
+    PlayerAvatar_SetUnk10(avatar, 0);
+    PlayerAvatar_SetUnk14(avatar, 0);
+    PlayerAvatar_SetState(avatar, state);
+    PlayerAvatar_SetGender(avatar, gender);
+    PlayerAvatar_SetTransitionFlags(avatar, 0);
+    PlayerAvatar_ClearUnk24ClearFlag2(avatar);
+    PlayerAvatar_SetUnk28(avatar, -1);
+    PlayerAvatar_SetUnk2C(avatar, -1);
+    PlayerAvatar_SetUnk8(avatar, 255);
+    PlayerAvatar_SetFlag1(avatar, TRUE);
+}
+
+static void PlayerAvatar_CreateMapObjectWithParams(PlayerAvatar *avatar, MapObjectManager *manager, u32 sprite, u32 direction, u32 x, u32 z) {
+    LocalMapObject *mapObject = MapObject_Create(manager, x, z, direction, sprite, 1, 1);
+    GF_ASSERT(mapObject != NULL);
+    MapObject_SetID(mapObject, 255);
+    MapObject_SetType(mapObject, 0);
+    MapObject_SetEventFlag(mapObject, 0);
+    MapObject_SetScriptID(mapObject, 0);
+    MapObject_SetParam(mapObject, 0, 0);
+    MapObject_SetParam(mapObject, 0, 1);
+    MapObject_SetParam(mapObject, 0, 2);
+    MapObject_SetXRange(mapObject, -1);
+    MapObject_SetYRange(mapObject, -1);
+    MapObject_SetFlagsBits(mapObject, (MapObjectFlagBits)(MAPOBJECTFLAG_UNK13 | MAPOBJECTFLAG_KEEP));
+    MapObject_ClearFlagsBits(mapObject, (MapObjectFlagBits)(MAPOBJECTFLAG_UNK8 | MAPOBJECTFLAG_UNK7));
+    MapObject_SetFlag29(mapObject, TRUE);
+    PlayerAvatar_SetMapObject(avatar, mapObject);
+}
+
+LocalMapObject *MapObjectManager_GetFirstActiveObjectWithMovement1(MapObjectManager *manager) {
+    s32 index = 0;
+    LocalMapObject *mapObject = NULL;
+    while (MapObjectManager_GetNextObjectWithFlagFromIndex(manager, &mapObject, &index, MAPOBJECTFLAG_ACTIVE)) {
+        if (MapObject_GetMovement(mapObject) == 1) {
+            break;
+        }
+    }
+    return mapObject;
+}
+
+static LocalMapObject *PlayerAvatar_GetActiveMapObjectWithMovement1(MapObjectManager *manager) {
+    LocalMapObject *mapObject = MapObjectManager_GetFirstActiveObjectWithMovement1(manager);
+    GF_ASSERT(mapObject != NULL);
+    return mapObject;
+}
+
+u32 PlayerAvatar_GetFacingDirection(PlayerAvatar *avatar) {
+    return MapObject_GetFacingDirection(PlayerAvatar_GetMapObject(avatar));
+}
+
+void PlayerAvatar_SetFacingDirection(PlayerAvatar *avatar, u32 direction) {
+    MapObject_SetFacingDirection(PlayerAvatar_GetMapObject(avatar), direction);
+}
+
+u32 PlayerAvatar_GetNextFacingDirection(PlayerAvatar *avatar) {
+    return MapObject_GetNextFacingDirection(PlayerAvatar_GetMapObject(avatar));
+}
+
+u32 PlayerAvatar_GetXCoord(PlayerAvatar *avatar) {
+    return MapObject_GetCurrentX(PlayerAvatar_GetMapObject(avatar));
```

Comment:
Alignment: Change to MapObject_GetXCoord or PlayerAvatar_GetCurrentX (I prefer former)